ncm-network/src/main/perl/nmstate.pm

```diff
@@ -296,6 +296,26 @@ sub generate_vip_config {
     return $iface_cfg;
 }
 
+# find vlan id from either the name or device.
+# i.e eth0.123 will return 123. by checking iface name and device
+# returns vlan id.
+sub get_vlanid {
+    my ($self, $iface, $device) = @_;
+    my $vlanid = $iface;
+    # a vlan interface can defined in two ways
+    # interface/name.vlanid/
+    # or interface/name/device=device.vlanid
+    # replace everything up-to and including . to get vlan id of the interface.
+    # favors ifacename.vlanid, then device.vlanid
+    $vlanid =~ s/^[^.]*.//;
+    # if vlanid is empty here, lets check if device has vlan id.
+    if ((! $vlanid) && ($device)) {
+        $vlanid = $device;
+        $vlanid =~ s/^[^.]*.//;
```

Comment:
same issue, make a small privtae sub so you can reuse the same  regex code. or make it i full blown sub?

```
sub _get_vlan_id {
    my ($self, $name) = @_;
    return if ($name =~ m/\.(\d+)$/) $1;  
}
```
(not tested)

Reply:
thanks, sorry for bad perl code :). updated.